src/utils/update/psc.to.beneficial.owner.type.mapper.ts

```diff
@@ -157,3 +157,16 @@ const natureOfControlTypeMap = new Map<string, NatureOfControlType>([
   [natureOfControl.SIGNIFICANT_INFLUENCE_OR_CONTROL_AS_FIRM, NatureOfControlType.SIGNIFICANT_INFLUENCE_OR_CONTROL]
 ]);
 
+export const mapPrivateAddress = (boData: BeneficialOwnersPrivateData, ch_reference: string) => {
+  if (!boData?.boPrivateData?.length) {
+    return;
+  }
+
+  for (const private_bo_data of boData.boPrivateData!) {
+    if (private_bo_data.hashedId === ch_reference) {
+      return private_bo_data.usualResidentialAddress
+        ? mapBOMOAddress(private_bo_data.usualResidentialAddress)
+        : mapBOMOAddress(private_bo_data.principalAddress);
```

Comment:
I think we should always have usualResidentialAddress and should never fall back on principalAddress (which I think is the service address)? It may still be safer to do so - just not expected?

Reply:
I don't think you can just rely on usualResidentialAddress being undefined to know to use principle address. Maybe we need to pass in something to say what kind of address to use.